### PR TITLE
remove magic numbers

### PR DESCRIPTION
**Pull request by LegitBaller0243**
## Description
Remove magic numbers issue: I updated the lines highlighted from #457, as well as two others (one was for creating a variable for standard atmospheric and the other was when the simulation checks if the grid is 3D) throughout the code. 

### Type of change

- [ ] Code cleaning

### Scope

- [x] This PR comprises a set of related changes with a common goal

If you cannot check the above box, please split your PR into multiple PRs that each have a common goal.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. 
Provide instructions so we can reproduce. 
Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:

* What computers and compilers did you use to test this:

## Checklist

- [ ] I have added comments for the new code
- [ ] I added Doxygen docstrings to the new code
- [ ] I have made corresponding changes to the documentation (`docs/`)
- [ ] I have added regression tests to the test suite so that people can verify in the future that the feature is behaving as expected
- [ ] I have added example cases in `examples/` that demonstrate my new feature performing as expected.
They run to completion and demonstrate "interesting physics"
- [ ] I ran `./mfc.sh format` before committing my code
- [ ] New and existing tests pass locally with my changes, including with GPU capability enabled (both NVIDIA hardware with NVHPC compilers and AMD hardware with CRAY compilers) and disabled
- [ ] This PR does not introduce any repeated code (it follows the [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) principle)
- [ ] I cannot think of a way to condense this code and reduce any introduced additional line count

### If your code changes any code source files (anything in `src/simulation`)

To make sure the code is performing as expected on GPU devices, I have:
- [ ] Checked that the code compiles using NVHPC compilers
- [ ] Checked that the code compiles using CRAY compilers
- [ ] Ran the code on either V100, A100, or H100 GPUs and ensured the new feature performed as expected (the GPU results match the CPU results)
- [ ] Ran the code on MI200+ GPUs and ensure the new features performed as expected (the GPU results match the CPU results)
- [ ] Enclosed the new feature via `nvtx` ranges so that they can be identified in profiles
- [ ] Ran a Nsight Systems profile using `./mfc.sh run XXXX --gpu -t simulation --nsys`, and have attached the output file (`.nsys-rep`) and plain text results to this PR
- [ ] Ran an Omniperf profile using `./mfc.sh run XXXX --gpu -t simulation --omniperf`, and have attached the output file and plain text results to this PR.
- [ ] Ran my code using various numbers of different GPUs (1, 2, and 8, for example) in parallel and made sure that the results scale similarly to what happens if you run without the new code/feature

### CODE REVIEW

File: src/pre_process/m_patches.fpp [hunk 1]
```diff
@@ -29,6 +29,7 @@ module m_patches
     ! ==========================================================================
 
     implicit none
+    integer, parameter :: grid_3D = 3
```
Comment (sbryngelson) [inline src/pre_process/m_patches.fpp:32]: grid_geometry = 3 means 3D_cylindrical, not 3D (general)

**Review by sbryngelson — CHANGES_REQUESTED** [review]
I find it hard to believe there were only a few magic numbers in the whole codebase

**Comment by sbryngelson** [comment]
A bunch of magic numbers here: https://github.com/LegitBaller0243/MFC/blob/e719c1631b75296d7b97c869d56546e65fae212c/src/simulation/m_rhs.fpp#L1901-L1946

**Comment by LegitBaller0243** [comment]
> A bunch of magic numbers here: https://github.com/LegitBaller0243/MFC/blob/e719c1631b75296d7b97c869d56546e65fae212c/src/simulation/m_rhs.fpp#L1901-L1946

Thank you for this and sounds good. I started with a couple, but needed a second opinion just to make sure I was on the right track. Will go back over code now.

**Comment by henryleberre** [comment]
@sbryngelson I would argue that all the following are also _magic numbers_:
- The boundary condition types.
- The model_eqns.
- The regular patch types.
- The acoustic supports.
- The immersed boundary patch types.

I feel like they should be named (using enums) and we just use the name in the code instead of the integer value. They are documented in https://mflowcode.github.io/documentation/md_case.html so it should be straightforward.

**Comment by henryleberre** [comment]
> > @sbryngelson I would argue that all the following are also _magic numbers_:
> > 
> > * The boundary condition types.
> > * The model_eqns.
> > * The regular patch types.
> > * The acoustic supports.
> > * The riemann solver types.
> > * The immersed boundary patch types.
> > * Probably many more
> >   …
> > 
> > I feel like they should be named (using enums) and we just use the name in the code instead of the integer value. They are documented in [mflowcode.github.io/documentation/md_case.html](https://mflowcode.github.io/documentation/md_case.html) so it should be straightforward.
> 
> I agree. I've thought about this before but never had the gumption to do something about it (breaking change of sorts).
> 
> What do you mean by
> 
> > I feel like they should be named (using enums)
> 
> .. Do you mean in the input files? If so, why is an enum necessary/useful here?

We could also name them in the input files but there isn't a very clean way of doing this. I was suggesting we could give them names in the Fortran code using a pattern a little like this:

m_patches.fpp:
```
module m_patches
    ...
    integer, parameter :: kLineSegmentPatch = 1
```

m_initial_condition.fpp:
```
if (patch_icpp(i)%geometry == kLineSegmentPatch) then
    call s_line_segment(i, patch_id_fp, q_prim_vf)
```

This is a lot more useful in places where the code is less readable like in the case of model_eqns. It also helps us keep better track of what integer maps to what type.

The alternative is to use [Fortran Enumerations](https://www.intel.com/content/www/us/en/docs/fortran-compiler/developer-guide-reference/2024-1/enumerations-and-enumerators.html).

**Comment by sbryngelson** [comment]
Yes this seems like a good idea.

**Comment by LegitBaller0243** [comment]
> > > @sbryngelson I would argue that all the following are also _magic numbers_:
> > > 
> > > * The boundary condition types.
> > > * The model_eqns.
> > > * The regular patch types.
> > > * The acoustic supports.
> > > * The riemann solver types.
> > > * The immersed boundary patch types.
> > > * Probably many more
> > >   …
> > > 
> > > I feel like they should be named (using enums) and we just use the name in the code instead of the integer value. They are documented in [mflowcode.github.io/documentation/md_case.html](https://mflowcode.github.io/documentation/md_case.html) so it should be straightforward.
> > 
> > 
> > I agree. I've thought about this before but never had the gumption to do something about it (breaking change of sorts).
> > What do you mean by
> > > I feel like they should be named (using enums)
> > 
> > 
> > .. Do you mean in the input files? If so, why is an enum necessary/useful here?
> 
> We could also name them in the input files but there isn't a very clean way of doing this. I was suggesting we could give them names in the Fortran code using a pattern a little like this:
> 
> m_patches.fpp:
> 
> ```
> module m_patches
>     ...
>     integer, parameter :: kLineSegmentPatch = 1
> ```
> 
> m_initial_condition.fpp:
> 
> ```
> if (patch_icpp(i)%geometry == kLineSegmentPatch) then
>     call s_line_segment(i, patch_id_fp, q_prim_vf)
> ```
> 
> This is a lot more useful in places where the code is less readable like in the case of model_eqns. It also helps us keep better track of what integer maps to what type.
> 
> The alternative is to use [Fortran Enumerations](https://www.intel.com/content/www/us/en/docs/fortran-compiler/developer-guide-reference/2024-1/enumerations-and-enumerators.html).


When I test the kLineSegmentPatch line, I keep getting this error: Error: Symbol 'klinesegmentpatch' at (1) has no IMPLICIT type. I followed the compiler instructions when setting up, and modules have use statements. What might be the reason causing this?

**Comment by sbryngelson** [comment]
I don't know i don't even see the code

File: src/pre_process/m_patches.fpp [hunk 2]
```diff
@@ -57,6 +58,30 @@ module m_patches
     real(kind(0d0)) :: x_centroid, y_centroid, z_centroid
     real(kind(0d0)) :: length_x, length_y, length_z
 
+    enum, bind(c)
+        enumerator :: kLineSegmentPatch = 1
```
Comment (sbryngelson) [inline src/pre_process/m_patches.fpp:62]: I'm unsure how this is different or helpful when compared to just declaring these as integer constant parameters in `common/m_constants.fpp`. @henryleberre had something in mind but honestly I'm not familiar with Fortran enumerations.

**Comment by sbryngelson** [comment]
closing due to inactivity

**Pull request closed without merging**